File: January 2021/26.01.2021/kadir.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
#include<cmath>
class Solution {
public:
    int getDecimalValue(ListNode* head) {
        int arr[31];
        int i = 0;
        int value = 0;
        int total = 0;
        ListNode* ptr = head;
        while(ptr){
            arr[i] = ptr->val;
            ptr = ptr->next;
            i++;
        }
        for(int a=0; a<=i;a++){
            if(arr[i-a] == 1){
                value = pow(2,a);
                total += value;
            }
            else{
                continue;
            }
        }
        return total/2;
    }
};

File: March 2021/14.03.2021/kadir.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode swapNodes(ListNode head, int k) {
        int a = 0;
        int first = 0;
        int second = 0;
        ListNode curr = head;
        while(curr != null){
            if(a == k-1){
                first = curr.val;
            }
            a++;
            curr = curr.next;
        }
        int secInd = a - k;
        a = 0;
        curr = head;
        while(curr != null){
            if(a == secInd){
                second = curr.val;
            }
            a++;
            curr = curr.next;
        }
        curr = head;
        a = 0;
        while(curr != null){
            if(a == k-1){
                curr.val = second;
            }            
            if(a == secInd){
                curr.val = first;
            }
            a++;
            curr = curr.next;
        }
        return head;
    }
}
